cmd/activator/main.go

```diff
@@ -126,16 +126,22 @@ func main() {
 	a := activator.NewRevisionActivator(kubeClient, servingClient, logger)
 	a = activator.NewDedupingActivator(a)
 
-	// Retry on 503's for up to 60 seconds. The reason is there is
-	// a small delay for k8s to include the ready IP in service.
+	// Retry on 502's, 503's and 504's for up to 60 seconds.
+	// The reason is there is a small delay for k8s to include the ready IP in service.
 	// https://github.com/knative/serving/issues/660#issuecomment-384062553
-	shouldRetry := activatorutil.RetryStatus(http.StatusServiceUnavailable)
+	// For 502's and 504's the reason is that some apps (like java) may take time to server requests
+	// even after the container is running
```

Comment:
I agree with @josephburnett here. It seems like that's exactly the job of a readinessProbe to take. Do you still see the same behavior with a readinessProbe in place? If yes, then that's a different bug to fix.

Our retry logic here should be as tight as possible and I think we're aiming to remove it completely since retries (especially for side-effecting actions) can have clunky semantics.